crates/e2e/tests/e2e/liquidity.rs

```diff
@@ -0,0 +1,409 @@
+
+#[tokio::test]
+#[ignore]
+async fn forked_node_zero_ex_liquidity_mainnet() {
+    run_forked_test_with_block_number(
+        zero_ex_liquidity,
+        std::env::var("FORK_URL_MAINNET")
+            .expect("FORK_URL_MAINNET must be set to run forked tests"),
+        FORK_BLOCK,
+    )
+    .await
+}
+
+async fn zero_ex_liquidity(web3: Web3) {
+    let mut onchain = OnchainComponents::deployed(web3.clone()).await;
+    let forked_node_api = web3.api::<ForkedNodeApi<_>>();
+
+    let [solver] = onchain.make_solvers_forked(to_wei(1)).await;
+    let [trader, zeroex_maker] = onchain.make_accounts(to_wei(1)).await;
+
+    let token_usdc = ERC20::at(
+        &web3,
+        "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48"
+            .parse()
+            .unwrap(),
+    );
+
+    let token_usdt = ERC20::at(
+        &web3,
+        "0xdac17f958d2ee523a2206206994597c13d831ec7"
+            .parse()
+            .unwrap(),
+    );
+
+    let zeroex = IZeroEx::deployed(&web3).await.unwrap();
+
+    let amount = to_wei_with_exp(5, 8);
+
+    // Give trader some USDC
+    let usdc_whale = forked_node_api.impersonate(&USDT_WHALE).await.unwrap();
+    tx!(usdc_whale, token_usdc.transfer(trader.address(), amount));
+
+    // Give 0x maker a bit more USDT
+    let usdt_whale = forked_node_api.impersonate(&USDT_WHALE).await.unwrap();
+    tx!(
+        usdt_whale,
+        token_usdt.transfer(zeroex_maker.address(), amount * 2)
+    );
+
+    // Approve GPv2 for trading
+    tx!(
+        trader.account(),
+        token_usdc.approve(onchain.contracts().allowance, amount)
+    );
+    tx!(
+        zeroex_maker.account(),
+        token_usdt.approve(zeroex.address(), amount * 2)
+    );
+
+    let order = OrderCreation {
+        sell_token: token_usdc.address(),
+        sell_amount: amount,
+        buy_token: token_usdt.address(),
+        buy_amount: amount - to_wei_with_exp(1, 8),
+        valid_to: model::time::now_in_epoch_seconds() + 300,
+        kind: OrderKind::Sell,
+        ..Default::default()
+    }
+    .sign(
+        EcdsaSigningScheme::Eip712,
+        &onchain.contracts().domain_separator,
+        SecretKeyRef::from(&SecretKey::from_slice(trader.private_key()).unwrap()),
+    );
+
+    let zeroex_api_port = {
+        let order = order.clone();
+        let chain_id = web3.eth().chain_id().await.unwrap().as_u64();
+        let weth_addr = onchain.contracts().weth.address();
+        let gpv2_addr = onchain.contracts().gp_settlement.address();
+        let zeroex_addr = zeroex.address();
+        let orders_handler = Arc::new(Box::new(move |query: &OrdersQuery| {
+            orders_query_handler(
+                query,
+                order.clone(),
+                zeroex_maker.clone(),
+                zeroex_addr,
+                gpv2_addr,
+                chain_id,
+                weth_addr,
+            )
+        }));
+
+        ZeroExApi::builder()
```

Comment:
This whole 0x API with Builder, etc seems a bit over-engineered in my opinion. Do you really need it to be query dependent? Shouldn't the API just take a list of orders in its constructor which it returns?

Reply:
True, initially, I was worried that some additional logic would be required for the API. For the current setup, this is not the case.